gptme/tools/base.py

```diff
@@ -240,3 +302,37 @@ def _to_xml(self) -> str:
         args = " ".join(self.args)
         args_str = "" if not args else f" args='{args}'"
         return f"<tool-use>\n<{self.tool}{args_str}>\n{self.content}\n</{self.tool}>\n</tool-use>"
+
+
+@dataclass(frozen=True)
+class APIToolUse(BaseToolUse):
+    tool: str
+    args: dict[str, str]
+
+    def execute(self, confirm: ConfirmFunc) -> Generator[Message, None, None]:
+        """Executes a tool-use tag and returns the output."""
+        # noreorder
+        from . import get_tool  # fmt: skip
+
+        tool = get_tool(self.tool)
+
+        if tool and (tool.execute_fn or tool.execute):
+            try:
+                if tool.execute_fn:
+                    ex = tool.execute_fn(self.args, confirm)
```

Comment:
```suggestion
                    content = self.args.get("content")
                    path = self.args.get("filepath") or self.args.get("path")
                    ex = tool.execute(content, [path] if path else [], confirm)
```

Reply:
I added a compat mode that should work for most of the tool like you suggested but I believe that we should migrate them all we proper argument when it's possible and it makes sense because it helps the model to use the tool properly.